README.md

```diff
@@ -2,31 +2,187 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Demonstrate access modifiers within an inheritance hierarchy
 
-## Introduction
+## Reviewing Access Modifiers
 
-A 1-2 sentence summary of what will be covered.
+Recall that an **access modifier** in Java
+controls the visibility of a class, field, constructor, or method.
+When an access modifier is not specified, the visibility defaults
+to the `package` level.
 
-## Topic 1
+| Modifier        | Class | Package  | Subclass In<br>Another Package | World |
+|-----------------|-------|----------|--------------------------------|-------|
+| public          | Y     | Y        | Y                              | Y     | 
+| protected       | Y     | Y        | Y  (within current object)     | N     | 
+| default/package | Y     | Y        | N                              | N     | 
+| private         | Y     | N        | N                              | N     | 
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+We'll explore how inheritance affects visibility in subclasses.
+Let's work with the sample class hierarchy shown below.
+Member access is denoted using one of the following symbols:
 
-## Topic 2
+- `-` private member
+- `+` public member
+- `#` protected member
+- `~` package/default member
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+![uml access modifier](https://curriculum-content.s3.amazonaws.com/6677/pillars/uml_access_modifier.png)
 
-## Conclusion
+The classes are contained in two packages `p1` and `p2`. 
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+- Package `p1` contains classes `Main`, `Superclass`, `ClassA`, and `SubclassB`.
+- Package `p2` contains classes `ClassC` and `SubclassD`.
+
+![directory structure](https://curriculum-content.s3.amazonaws.com/6677/pillars/directory_structure.png)
+
+The `Main` driver creates an instance of each class to show the object state due to inheritance.
+
+```java
+package p1;
+
+import p2.ClassC;
+import p2.SubclassD;
```

Comment:
This might be a good place to remind students that we need to import these two classes since they live in a different package.

Reply:
added a comment